File: src/main/java/io/github/legacy_fabric_community/serialization/codec/VanillaRegistryCodecs.java
package io.github.legacy_fabric_community.serialization.codec;

import com.mojang.serialization.Codec;

import net.minecraft.block.Block;
import net.minecraft.item.Item;

public interface VanillaRegistryCodecs {
    Codec<Block> BLOCK = (Codec<Block>) Block.REGISTRY;
    Codec<Item> ITEM = (Codec<Item>) Item.REGISTRY;
}

File: shadowed-config-registries/src/main/scala/io/github/legacy_fabric_community/shadowedconfig/registries/VanillaRegistryCodecs.scala
package io.github.legacy_fabric_community.shadowedconfig.registries

import com.mojang.serialization.Codec
import net.minecraft.block.Block
import net.minecraft.item.Item

@SuppressWarnings(Array("unchecked")) object VanillaRegistryCodecs {
	val BLOCK: Codec[Block] = Block.REGISTRY.asInstanceOf[Codec[Block]]
	val ITEM: Codec[Item] = Item.REGISTRY.asInstanceOf[Codec[Item]]
}
